### PR TITLE
Add info and join

### PR DESCRIPTION
**Pull request by barrachri**
## What do these changes do?

Add support for `docker info` and `docker swarm join`.

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `changes` folder
  * name it `<issue_id>.<type>` for example (588.bug)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: aiodocker/swarm.py [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-                   force_new_cluster: bool=False,
-                   swarm_spec: Optional[Dict]=None
-                   ) -> str:
+    async def init(
+        self,
+        *,
+        advertise_addr: str=None,
+        listen_addr: str="0.0.0.0:2377",
+        force_new_cluster: bool=False,
+        swarm_spec: Dict=None
```
Comment (achimnol) [inline aiodocker/swarm.py]: I'd prefer more generic `typing.Mapping` or `typing.MutableMapping` here.
Q: Have we agreed with remove `Optional` for nullable parameters? (Maybe I have missed the conversation due to inactivity...)

Comment (barrachri) [reply aiodocker/swarm.py]: ```Dict=None is like Optional[Dict]```, I'll change the type with mapping 👍

Comment (achimnol) [reply aiodocker/swarm.py]: Ah, so `=None` implies `Optional` in this case. Got it. :+1:

File: aiodocker/swarm.py [hunk 2]
```diff
@@ -59,13 +58,57 @@ def __init__(self, docker):
 
         return response
 
+    async def join(
+            self,
+            *,
+            remote_addrs: List[str],
```
Comment (achimnol) [inline aiodocker/swarm.py]: `List` -> `Iterable`

File: aiodocker/system.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from typing import Dict
+
+
+class DockerSystem(object):
+    def __init__(self, docker):
+        self.docker = docker
+
+    async def info(self) -> Dict:
```
Comment (achimnol) [inline aiodocker/system.py]: `Dict` -> `Mapping`

File: aiodocker/swarm.py [hunk 4]
```diff
@@ -59,13 +58,57 @@ def __init__(self, docker):
...
+
+            remote_addrs
+                Addresses of manager nodes already participating in the swarm.
+
+            join_token
+                Secret token for joining this swarm.
+        """
+
+        data = {
+            "RemoteAddrs": remote_addrs,
```
Comment (achimnol) [inline aiodocker/swarm.py]: If we change the parameter type to `Iterable`, it would be safer to wrap this with `tuple()` or `list()`.

**Comment by achimnol** [comment]
:+1:

**Pull request merged**